server/src/main/java/io/seata/server/store/db/DatabaseTransactionStoreManager.java

```diff
@@ -184,17 +187,17 @@ public GlobalSession readSession(String xid, boolean withBranchSessions) {
         for (int i = 0; i < statuses.length; i++) {
             states[i] = statuses[i].getCode();
         }
-        List<GlobalSession> globalSessions = new ArrayList<GlobalSession>();
+        List<GlobalSession> globalSessions = new ArrayList<>();
         //global transaction
         List<GlobalTransactionDO> globalTransactionDOs = logStore.queryGlobalTransactionDO(states, logQueryLimit);
         if (CollectionUtils.isEmpty(globalTransactionDOs)) {
             return null;
         }
-        for (GlobalTransactionDO globalTransactionDO : globalTransactionDOs) {
-            List<BranchTransactionDO> branchTransactionDOs = logStore.queryBranchTransactionDO(
-                globalTransactionDO.getXid());
-            globalSessions.add(getGlobalSession(globalTransactionDO, branchTransactionDOs));
-        }
+        List<String> xids = globalTransactionDOs.stream().map(GlobalTransactionDO::getXid).collect(Collectors.toList());
+        List<BranchTransactionDO> branchTransactionDOs = logStore.queryBranchTransactionDO(xids);
+        Map<String, List<BranchTransactionDO>> branchTransactionDOsMap = branchTransactionDOs.stream()
+            .collect(Collectors.groupingBy(BranchTransactionDO::getXid, LinkedHashMap::new, Collectors.toList()));
+        globalTransactionDOs.stream().forEach(globalTransactionDO -> globalSessions.add(getGlobalSession(globalTransactionDO, branchTransactionDOsMap.get(globalTransactionDO.getXid()))));
```

Comment:
Looks too complex in one line, maybe can expand body as `{...}`.

Reply:
Adjust the code structure to make it look sharper.
 In addition, the variable name branchTransactionDOs already exists, and branchTransactionDOsMap.get is not well named.